src/main/io/osd.c

```diff
@@ -4283,264 +4352,490 @@ static void osdUpdateStats(void)
+    char buff[12];
+    uint8_t multiValueXOffset = 0;
+
+    const float max_gforce = accGetMeasuredMaxG();
+    const acc_extremes_t *acc_extremes = accGetMeasuredExtremes();
+    const float acc_extremes_min = acc_extremes[Z].min;
+    const float acc_extremes_max = acc_extremes[Z].max;
+
+    if (!osdDisplayIsHD())
+        displayWrite(osdDisplayPort, col, row, "MAX G-FORCE");
+    else
+        displayWrite(osdDisplayPort, col, row, "MAX/MIN Z/MAX Z G-FORCE");
+        
+    tfp_sprintf(buff, ": ");
+    osdFormatCentiNumber(buff + 2, max_gforce * 100, 0, 2, 0, 3, false);
+    
+    if (!osdDisplayIsHD()) {
+        displayWrite(osdDisplayPort, statValX, row++, buff);
+
+        displayWrite(osdDisplayPort, col, row, "MIN/MAX Z G-FORCE");
+        memset(buff, '\0', strlen(buff));
+        tfp_sprintf(buff, ": ");
+    } else {
+        strcat(osdFormatTrimWhiteSpace(buff), "/");
+    }
+    multiValueXOffset = strlen(buff);
+    osdFormatCentiNumber(buff + multiValueXOffset, acc_extremes_min * 100, 0, 2, 0, 4, false);
+    osdLeftAlignString(buff);
+    strcat(osdFormatTrimWhiteSpace(buff), "/");
+    
+    multiValueXOffset = strlen(buff);
+    osdFormatCentiNumber(buff + multiValueXOffset, acc_extremes_max * 100, 0, 2, 0, 3, false);
+    osdLeftAlignString(buff);
+    displayWrite(osdDisplayPort, statValX, row++, buff);
+
+    return row;
+}
+
+uint8_t drawStat_DisarmMethod(uint8_t col, uint8_t row, uint8_t statValX) {
+    const char * disarmReasonStr[DISARM_REASON_COUNT] = { "UNKNOWN", "TIMEOUT", "STICKS", "SWITCH", "SWITCH", "KILLSW", "FAILSAFE", "NAV SYS", "LANDING"};
+
+    displayWrite(osdDisplayPort, col, row, "DISARMED BY");
+    displayWrite(osdDisplayPort, statValX, row, ": ");
+    displayWrite(osdDisplayPort, statValX + 2, row++, disarmReasonStr[getDisarmReason()]);
+
+    return row;
+}
+
+static void osdShowStats(bool isSinglePageStatsCompatible, uint8_t page)
+{
+    const char * statsHeader[2] = {"*** STATS   1/2 -> ***", "*** STATS   <- 2/2 ***"};
+    uint8_t row = 1;  // Start one line down leaving space at the top of the screen.
+
+    const uint8_t statNameX = (osdDisplayPort->cols - (osdDisplayIsHD() ? 41 : 28)) / 2;
+    const uint8_t statValuesX = osdDisplayPort->cols - statNameX - (osdDisplayIsHD() ? 15 : 11);
+    
+    if (page > 1)
+        page = 0;
+
+    displayBeginTransaction(osdDisplayPort, DISPLAY_TRANSACTION_OPT_RESET_DRAWING);
+    displayClearScreen(osdDisplayPort);
+
+    if (isSinglePageStatsCompatible) {
+        char buff[20];
+        tfp_sprintf(buff, "*** STATS ");
+#ifdef USE_BLACKBOX
+#ifdef USE_SDCARD
+        if (feature(FEATURE_BLACKBOX)) {
+            int32_t logNumber = blackboxGetLogNumber();
+            if (logNumber >= 0)
+                tfp_sprintf(buff + strlen(buff), " %c%05" PRId32 " ", SYM_BLACKBOX, logNumber);
+            else
+                tfp_sprintf(buff + strlen(buff), " %c ", SYM_BLACKBOX);
+        }
+#endif
+#endif
+        strcat(buff, "***");
```

Comment:
Looks like we have a buffer overflow here:
```
char buff[20];
         1         2
12345678901234567890
*** STATS  Xddddd ***@
```
where `X` is  `SYM_BLACKBOX` and `@` is the trailing NUL byte. Note also double spaces between STATS and symbol

Reply:
Nice spot. I've increased the buffer. I added a little overflow, as I may include the three characters from the start of the file in a future update.